### DISCUSSION TITLE
Alternative locations for toolbars

### DISCUSSION THREAD

**Discussion by IntelOrca** [OpenRCT2/OpenRCT2#15643] [Feature requests] [answered]
Some people may want to left align, centre or right align their top and bottom toolbars. Perhaps even a custom location, particularly if the application window is spread across multiple monitors.

**Comment by IntelOrca** [comment]
@AaronVanGeffen is this what you wanted?
![scr54](https://cloud.githubusercontent.com/assets/1482259/5331949/ece69fb8-7e38-11e4-9d7c-45f8577eb332.png)

**Comment by vzybilly** [comment]
If the toolbars are made more dynamic in location, might it be possible to add the ability to have an 'advance placement' option where you can click and drag each button to where you want it in the screen while in the toolbar advance placement selection. maybe even add your own buttons by listing the windows then tabs in the windows, allowing for a new button to go directly to a tab of a window? If this is all done, I easily see people being happy by moving all the buttons into their own groupings and at the locations of their choice... I'll see if I can't edit an image together to show more of what I mean...

**Comment by IntelOrca** [comment]
@vzybilly It would be interesting to see how many people would want this feature. The user might be overwhelmed by the complexity of the toolbar if it is so customisable. It would also be harder to maintain in the code base too. I also think that players are more likely, going to want / use hotkeys to get to particular windows and tabs, and that the players who don't use hotkeys aren't bothered about a button to access a particular tab quicker.

**Comment by vzybilly** [comment]
True, I was thinking it would all be in a toolbar settings window that has a drop down box(?) that has the options of "Top Left", "Top Center", "Top Right", "Above Event Bar"(?), "Custom" and when the custom is selected it says to drag and drop the tool bar buttons. as for ease of code, maybe just a X,Y pointer attached to each button for drawing? I also just thought of a possible grid where the buttons snap in?
Sorry if all this is way to much but I do like to see highly customize-able things...

Also, here is my poorly edited picture showing a placement that I might use, the ones on the event bar(?) could be moved to the right side of the screen but this is just my thoughts.
![ece69fb8-7e38-11e4-9d7c-45f8577eb332](https://cloud.githubusercontent.com/assets/6563474/5332349/3df82d76-7e21-11e4-8d3a-08bc0f3bc9a2.png)

**Comment by AaronVanGeffen** [comment]
@IntelOrca Yes, that looks a lot like what I had in mind! I'd be very happy with just that.

Expanding on it, maybe it'd be nice to have the option to show the main toolbar vertically on either side of the screen. I don't think I'd personally use that, though.

**Comment by Gymnasiast** [comment]
I have no problem with the toolbars as they currently are, but if this feature gets implemented, I would like to be able to move the top toolbar to the left of the screen, with the pause button on top and all the other buttons vertically arranged underneath it. (Which I think is what @AaronVanGeffen intended with his last post.)

Generally speaking, I wouldn't complicate the toolbar placement and the code base too much for a nice-to-have feature. If I had to decide on my own I would provide the following options:

Top/main toolbar: top / left / right / bottom
Bottom/status bar: bottom / top
And possibly an additional option (if the resolution is high enough) for a unified toolbar on either the top or bottom of the screen.

**Comment by RollingStar** [comment]
Building an in-game way to re-arrange the buttons might be overkill. Assuming people do want to re-arrange their buttons, a list of X,Y co-ordinates for each button could be placed in the config file instead.

And most people are probably fine with the buttons staying in the same spot they've always been.

**Comment by Ryder17z** [comment]
It would be nice if you could rearrange the toolbar locations like in Need For Speed Porsche.
(You hold down a key combo, game gets paused and you can move everything around as you want, hit enter to save or esc to discard the changes)

**Comment by janisozaur** [comment]
See also https://www.reddit.com/r/rct/comments/3xuxlz/orct_moving_menus_for_multiple_monitors/

I imagine we could possibly make the toolbar follow the cursor so it would be available on the screen you're currently using. If you move the cursor to another screen, the toolbar will do so too.

**Comment by marijnvdwerf** [comment]
What if you would make the toolbar order a config file option? No UI for customizing it, except maybe for enabling/displaying features.

```
top_toolbar = pause | fastforward | file_menu | network | cheats | debug | - | zoom_out | zoom_in | rotate | view_menu | map | ~ | news | guests | staff | park | rides | research | finances | - | construct_ride | path | scenery | water | land | clear_scenery
```

Would be parsed ignoring whitespace. Vertical bars would seperate the different items. `-` would be simple fixed-width whitespace, ~ would be fluid width whitespace, value gets calculated based on current screen width, minus other content, divided by number of fluid whitespace items. If there is no fluid whitespace in the toolbar, it will be appended at the end (making the toolbar left aligned)

**Comment by TheLarry98** [comment]
This would be a very good feature :)

**Comment by AaronVanGeffen** [comment]
After ten years, I might just have a go at implementing this :)

**Comment by AaronVanGeffen** [comment, marked as answer]
This has now been implemented in develop: #21989
